pkg/static/login.js

```diff
@@ -747,15 +750,48 @@ import "./login.scss";
     function do_hostkey_verification(data) {
         const key_db = get_known_hosts_db();
         const key = data["host-key"];
-        const key_key = key.split(" ")[0];
-        const key_type = key.split(" ")[1];
+        const [key_key, key_type, key_value] = key.split(" ");
+        let db_key = key_db[key_key];
+
+        // did we see this host before?
+        if (db_key) {
+            // exact same key → good
+            if (key_db[key_key] == key) {
+                converse(data.id, data.default);
+                return;
+            }
 
-        if (key_db[key_key] == key) {
-            converse(data.id, data.default);
-            return;
+            /* transition case: cockpit-ssh stored full public keys in the DB, like
+            * "myhost ecdsa-sha2-nistp256 AAAA[..]ToM=\n"; ssh(1) does not show them, only fingerprints, so
+            * cockpit-beiboot only responds with fingerprints as well; they look like "myhost ED25519 SHA256:JVG[..]xs"
+            * check if we can match the given fingerprint to an existing entry */
+            const [db_key_type, db_key_value] = db_key.split(" ").slice(1);
+            if (key_value.startsWith('SHA256:') && db_key_value && !db_key_value.includes(':')) {
+                // cockpit-ssh and ssh use different notation: ED25519 vs. ssh-ed25519
```

Comment:
```c
⬢[lis@toolbox openssh-portable]$ git grep '\.name =\|\.shortname ='
...
ssh-dss.c:      /* .name = */           "ssh-dss",
ssh-dss.c:      /* .shortname = */      "DSA",
ssh-dss.c:      /* .name = */           "ssh-dss-cert-v01@openssh.com",
ssh-dss.c:      /* .shortname = */      "DSA-CERT",
ssh-ecdsa-sk.c: /* .name = */           "sk-ecdsa-sha2-nistp256@openssh.com",
ssh-ecdsa-sk.c: /* .shortname = */      "ECDSA-SK",
ssh-ecdsa-sk.c: /* .name = */           "sk-ecdsa-sha2-nistp256-cert-v01@openssh.com",
ssh-ecdsa-sk.c: /* .shortname = */      "ECDSA-SK-CERT",
ssh-ecdsa-sk.c: /* .name = */           "webauthn-sk-ecdsa-sha2-nistp256@openssh.com",
ssh-ecdsa-sk.c: /* .shortname = */      "ECDSA-SK",
ssh-ecdsa.c:    /* .name = */           "ecdsa-sha2-nistp256",
ssh-ecdsa.c:    /* .shortname = */      "ECDSA",
ssh-ecdsa.c:    /* .name = */           "ecdsa-sha2-nistp256-cert-v01@openssh.com",
ssh-ecdsa.c:    /* .shortname = */      "ECDSA-CERT",
ssh-ecdsa.c:    /* .name = */           "ecdsa-sha2-nistp384",
ssh-ecdsa.c:    /* .shortname = */      "ECDSA",
ssh-ecdsa.c:    /* .name = */           "ecdsa-sha2-nistp384-cert-v01@openssh.com",
ssh-ecdsa.c:    /* .shortname = */      "ECDSA-CERT",
ssh-ecdsa.c:    /* .name = */           "ecdsa-sha2-nistp521",
ssh-ecdsa.c:    /* .shortname = */      "ECDSA",
ssh-ecdsa.c:    /* .name = */           "ecdsa-sha2-nistp521-cert-v01@openssh.com",
ssh-ecdsa.c:    /* .shortname = */      "ECDSA-CERT",
ssh-ed25519-sk.c:       /* .name = */           "sk-ssh-ed25519@openssh.com",
ssh-ed25519-sk.c:       /* .shortname = */      "ED25519-SK",
ssh-ed25519-sk.c:       /* .name = */           "sk-ssh-ed25519-cert-v01@openssh.com",
ssh-ed25519-sk.c:       /* .shortname = */      "ED25519-SK-CERT",
ssh-ed25519.c:  /* .name = */           "ssh-ed25519",
ssh-ed25519.c:  /* .shortname = */      "ED25519",
ssh-ed25519.c:  /* .name = */           "ssh-ed25519-cert-v01@openssh.com",
ssh-ed25519.c:  /* .shortname = */      "ED25519-CERT",
ssh-rsa.c:      /* .name = */           "ssh-rsa",
ssh-rsa.c:      /* .shortname = */      "RSA",
ssh-rsa.c:      /* .name = */           "ssh-rsa-cert-v01@openssh.com",
ssh-rsa.c:      /* .shortname = */      "RSA-CERT",
ssh-rsa.c:      /* .name = */           "rsa-sha2-256",
ssh-rsa.c:      /* .shortname = */      "RSA",
ssh-rsa.c:      /* .name = */           "rsa-sha2-512",
ssh-rsa.c:      /* .shortname = */      "RSA",
ssh-rsa.c:      /* .name = */           "rsa-sha2-256-cert-v01@openssh.com",
ssh-rsa.c:      /* .shortname = */      "RSA-CERT",
ssh-rsa.c:      /* .name = */           "rsa-sha2-512-cert-v01@openssh.com",
ssh-rsa.c:      /* .shortname = */      "RSA-CERT",
ssh-xmss.c:     /* .name = */           "ssh-xmss@openssh.com",
ssh-xmss.c:     /* .shortname = */      "XMSS",
ssh-xmss.c:     /* .name = */           "ssh-xmss-cert-v01@openssh.com",
ssh-xmss.c:     /* .shortname = */      "XMSS-CERT",
```

Reply:
Meh.. But I just realized that we don't really actually have to do this comparison. If we assume collision resistance (SHA256 's only job), then we can either be much more lenient about this and do a lower-case substring match, or even just  drop it completely -- the hash of a different key type is going to be different. If not, the world has much bigger problems :cold_sweat:

Reply:
This comparison is gone now.

Reply:
There's no fingerprint conversion or all this type magic any more.